api/lib/cat/assessment.js

```diff
@@ -0,0 +1,119 @@
+Set.prototype.union = function(setB) {
+  const union = new Set(this);
+  for (const elem of setB) {
```

Comment:
Je préfère laisser un for..each car c'est l'implem exacte donnée par la doc de Set.

Par ailleurs, ça permet d'avoir une implem très proche entre ```union``` et ```difference``` (dans le second cas, on peut aussi utiliser reduce, mais ce serait incompréhensible).